benchmarks/benchmarks/sparse_csgraph_matching.py

```diff
@@ -0,0 +1,21 @@
+import scipy.sparse
+
+try:
+    from scipy.sparse.csgraph import maximum_bipartite_matching
+except ImportError:
+    pass
+
+from .common import Benchmark
+
+
+class MaximumBipartiteMatching(Benchmark):
+    params = [[1000, 5000, 10000], [0.01, 0.1, 0.2]]
```

Comment:
this takes forever to run. I propose to change it to `params = [[1000, 2500], [0.01, 0.1]]`. That still takes about 8 seconds on my Macbook, with result:
```
$ asv run --bench MaximumBipartiteMatching.time_maximum_bipartite_matching --python=same
· Discovering benchmarks
· Running 1 total benchmarks (1 commits * 1 environments * 1 benchmarks)
[  0.00%] ·· Benchmarking existing-py_Users_rgommers_anaconda3_bin_python
[ 50.00%] ··· Running (sparse_csgraph_matching.MaximumBipartiteMatching.time_maximum_bipartite_matching--).
[100.00%] ··· ...ng.MaximumBipartiteMatching.time_maximum_bipartite_matching                 ok
[100.00%] ··· ====== ========= ============
              --            density        
              ------ ----------------------
                n       0.01       0.1     
              ====== ========= ============
               1000   268±3μs    372±3μs   
               2500   779±6μs   4.31±0.3ms 
              ====== ========= ============
```

Reply:
Or if you want to keep a 3x3 matrix, add smaller values (say n=250, density=0.025)

Reply:
Updated.

I actually took the threshold for acceptable run time from [this comment](https://github.com/scipy/scipy/pull/10296#issuecomment-502232642) (might be worth including a note on what's acceptable in [the devdocs](https://scipy.github.io/devdocs/dev/contributor/benchmarking.html#writing-benchmarks)?):

>  I have kept the cost matrix sizes small in order to keep the running time to ~20 seconds on my laptop.

So let me double-check that the original benchmark did what I wanted it to do. Will comment or add a follow-up PR if anything's off.

Reply:
Just for the record, I ran these again, and indeed with the original setup, they take forever (a few minutes). Notably, `time_maximum_bipartite_matching` itself completes immediately, but setting up the input data (through `scipy.sparse.rand`) is what ends up taking all the time.

Reply:
I changed the setup code in https://github.com/scipy/scipy/pull/10641/commits/bd6871fa3f9c39227a02cf3e9a37e5515d27d900 so that setup is now reasonably fast, and the matchings themselves more interesting.